osu.Game/Graphics/UserInterface/OsuDropdown.cs

```diff
@@ -30,6 +31,12 @@ public partial class OsuDropdown<T> : Dropdown<T>, IKeyBindingHandler<GlobalActi
 
         protected override DropdownMenu CreateMenu() => new OsuDropdownMenu();
 
+        public OsuDropdown()
+        {
+            if (Header is OsuDropdownHeader osuHeader)
+                osuHeader.Dropdown = this;
+        }
```

Comment:
The header needs access to the dropdown itself to read the menu state, and I chose to pass it down this way to avoid too many LoCs, since there are inheritors of `OsuDropdownHeader` and I don't think it's wise to update each constructor just to do that.

I can also use DI but I think it's not worth it, unless this is too ugly and I can't see it.